### ISSUE TITLE
Module Federation + base url

### ISSUE THREAD

**Issue by terragady** [originjs/vite-plugin-federation#580]
Hi, starting from Vite5 the module federation plugin was not working properly with the Vite.

We are using baseUrl, lets say `/base/url` and we have this setting in vite config: 
`base: "/base/url"`
which is fine but the whole app does not work in dev mode (local server) and we figured out it needed setting `server: { origin: "/base/url }`  and then it starts and module federation works properly but the problem starts with images now, they are not found because double base url is applied like `http://localhost:8080/base/url/base/url/assets/images/image.png
`
so we have removed the origin and tried other fix and we had to use proxy for node_modules like this: 
      `  '/node_modules': 'http://localhost:8080/base/url/',`

but my gut feeling says it is a wrong solution and hardcoding port and url here but it works fine with dev, build, federation and images. Anyone knows what is best solution here and what we are doing wrong/ or good? 🙂

**Comment by an501920078** [comment]
try add server.origin

**Comment by terragady** [comment]
I am doing this but then the assets does not load as they have double base url

**Comment by an501920078** [comment]
Yes, I agree with you, and I also find that the new URL is problematic if you add server.origin

**Comment by AdityaSP** [comment]
Hi, did you find a solution to this? Can you please explain the proxy for node modules?

**Comment by an501920078** [comment]
![53BD5151-F419-4c12-8B0F-BB9D86C0B725](https://github.com/user-attachments/assets/186b8bf7-65d4-4e4f-b49e-756456f9a9f9)
It's not good, but it has to be

**Comment by wuchunxu** [comment]
Since the base URL works well in the production mode, we can use enviroment variables to avoid this problem in development mode.
```txt
# .env.development
VITE_BASE_URL='/'
```
```txt
# .env.production
VITE_BASE_URL='/base/url'
```
```js
// vite.config.js
import { defineConfig, loadEnv } from 'vite'
export default defineConfig(({ mode }) => {
  const env = loadEnv(mode, process.cwd(), '')
  return {
    base: env.VITE_BASE_URL,
    // ...
  }
}
```

**Comment by EduardvonBriesen** [comment]
This configuration worked for me, without `server.origin`

```typescript
// vite.config.ts
server: {
  proxy: {
    [`/node_modules`]: {
      target: `http://localhost:${env.PORT}${env.VITE_BASEPATH}`,
    },
  }
}
```

**Comment by EduardvonBriesen** [comment]
This solution worked for me, without `server.origin`

```typescript
server: { 
   proxy: {
     [`/node_modules`]: {
       target: env.VITE_BASEPATH,
     },
   }
}
```

**Comment by swag-code-hub** [comment]
Any solution of this with out using proxy for node modules?

**Comment by mstefanak-cen58378** [comment]
Hi, any updates on this please?

**Comment by ksv90** [comment]
> Hi, any updates on this please?

As @wuchunxu already described, you can use environment variables. In my case, I don't use the additional VITE_BASE_URL variable and statically add base only for deploy.

`base: NODE_ENV === 'production' ? '/url' : '',`

**Comment by jtweeks** [comment]
I am having the same issue, baseUrl appended after assets in React, Vite federated app.  Local is fine, happening on deployed app.
